umm-lib/src/cmr/umm/related_url_helper.clj

```diff
@@ -86,3 +87,28 @@
   "Returns a sequence of atom links for the given related urls"
   [related-urls]
   (map related-url->atom-link related-urls))
+
+(defn- url-encoded?
+  "Test to see if url is encoded."
+  [url]
+  (not (= url (codec/url-decode url))))
+
+(defn- encode-url
+  "Encode the query portion of the url."
+  [url]
+  (let [split-url (string/split url #"\?")
+        base-url (first split-url)
+        params (apply str (rest split-url))]
+    (if (not-empty params)
+      (->> params
+           codec/form-decode
+           codec/form-encode
+           (str base-url "?"))
+      url)))
+
+(defn related-url->encoded-url
```

Comment:
This function is perfect for unit testing.

Add unit tests testing a bunch of different cases - not URL encoded, spaces, +, ? in the value of a query parameter, etc.

Unit tests would go in a namespace:

umm-lib/**test**/cmr/umm/related_url_helper.clj